src/test/ui/traits/trait-bounds-same-crate-name.rs

```diff
@@ -0,0 +1,21 @@
+// aux-build:crate_a1.rs
+// aux-build:crate_a2.rs
```

Comment:
There's one new test for the case where the diagnostic is correct. Do we also want testcases that verify that the diagnostic is not emitted?
E.g. there exists a trait with same path but it's not implemented for the struct.

Reply:
Negative tests are always good to have, particularly because incorrect suggestions are sometimes worse than no suggestions. Can you create one for the case you mention?

Reply:
I'm adding a few cases:
- A type that doesn't implement any of the two traits
- A type that doesn't implement any of the two traits, but a different variant of the type implements the wrong trait
- A type that doesn't implement any of the two traits, but a different variant of the type implements the correct trait